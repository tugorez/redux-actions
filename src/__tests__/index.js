import { createActions, createReducer } from '../index';

describe('createActionsActions', () => {
  it('should createActions actions', () => {
    const prefix = '@prefix';
    const types = ['RANDOM_ACTION1', 'RANDOM_ACTION2'];
    const actions = createActions(prefix, types);
    expect(actions.randomAction1()).toMatchSnapshot();
    expect(actions.randomAction2()).toMatchSnapshot();
  });

  it('should create a valid reducer', () => {
    const prefix = '@calculator';
    const actions = createActions(prefix, ['ADD', 'SUB']);
    const initial = 0;
    const reducer = createReducer(initial, {
      [actions.add]: (state, { payload = 1 }) => state + payload,
      [actions.sub]: (state, { payload = 1 }) => state - payload,
    });
    expect(reducer(undefined, actions.add())).toMatchSnapshot();
    expect(reducer(undefined, actions.sub())).toMatchSnapshot();
  });
});
