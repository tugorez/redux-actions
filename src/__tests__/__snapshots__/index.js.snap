// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createActionsActions should create a valid reducer 1`] = `1`;

exports[`createActionsActions should create a valid reducer 2`] = `-1`;

exports[`createActionsActions should createActions actions 1`] = `
Object {
  "type": "@prefix/RANDOM_ACTION1",
}
`;

exports[`createActionsActions should createActions actions 2`] = `
Object {
  "type": "@prefix/RANDOM_ACTION2",
}
`;
